src/sim_engine/filesystem.rs

```diff
@@ -15,14 +15,17 @@ pub struct SimFilesystem {
     pub name: String,
     pub mount_point: String,
     pub quota_size: u64,
+    pub ancestor: Option<Uuid>,
 }
+
 impl SimFilesystem {
     pub fn new_filesystem(name: &str, mount_point: &str, quota_size: Option<u64>) -> SimFilesystem {
         SimFilesystem {
             name: name.to_owned(),
             uuid: Uuid::new_v4(),
             mount_point: mount_point.to_owned(),
             quota_size: quota_size.unwrap_or(DEFAULT_FILESYSTEM_QUOTA_SIZE),
+            ancestor: None,
```

Comment:
This and other ancestors should be changed to ```parent``` or possibly ```nearest_ancestor``` since parent's can go away.